applications/FluidDynamicsApplication/custom_strategies/strategies/fs_strategy.h

```diff
@@ -228,164 +154,163 @@ class FSStrategy : public SolvingStrategy<TSparseSpace,TDenseSpace,TLinearSolver
-
-        }
-        else
-        {
+            KRATOS_WARNING_IF("FSStrategy", !converged && echo_level > 0) << "Predictor-correctior iterations did not converge." << std::endl;
+        } else {
             // Solve for fractional step velocity, then update pressure once
-            NormDp = this->SolveStep();
+            norm_dp = this->SolveStep();
+            // If not doing predictor corrector iterations, norm_dp will
+            // typically be "large" since we are not iterating on pressure.
+            // It makes no sense to report that the iteration didn't converge
+            // based on this.
+            converged = true;
         }
 
-        if (mReformDofSet)
-            this->Clear();
-
-
+        // Calculate reactions
+        if (mCalculateReactionsFlag) {
+            CalculateReactions();
+        }
 
-        return NormDp;
+        return converged;
     }
 
-    bool SolveSolutionStep() override
+    void FinalizeSolutionStep() override
     {
-        double norm_dp = this->Solve();
-        /* If not doing predictor corrector iterations, norm_dp will
-         * typically be "large" since we are not iterating on pressure.
-         * It makes no sense to report that the iteration didn't converge
-         * based on this.
-         */
-        return mPredictorCorrector ? this->CheckPressureConvergence(norm_dp) : true;
+        if (mReformDofSet) {
+            this->Clear();
+        }
     }
 
+    //TODO: Move to private section as soon as we remove the Python exposure
+    /**
+     * @brief Calculates the reactions
+     * This methods calculates the reactions of the momentum equation.
+     * These are computed as minus the RHS and saved in the REACTION variable
+     */
+virtual void CalculateReactions()
+{
+    auto &r_model_part = BaseType::GetModelPart();
+    auto &r_process_info = r_model_part.GetProcessInfo();
 
-    virtual void CalculateReactions()
-    {
-
-        ModelPart& rModelPart = BaseType::GetModelPart();
-        ProcessInfo& rCurrentProcessInfo = rModelPart.GetProcessInfo();
-
-        // Set fractional step index to the momentum equation step
-        int OriginalStep = rCurrentProcessInfo[FRACTIONAL_STEP];
-        rCurrentProcessInfo.SetValue(FRACTIONAL_STEP,1);
-
-#pragma omp parallel
-        {
-            ModelPart::NodeIterator NodesBegin;
-            ModelPart::NodeIterator NodesEnd;
-            OpenMPUtils::PartitionedIterators(rModelPart.Nodes(),NodesBegin,NodesEnd);
+    // Set fractional step index to the momentum equation step
+    const int original_step = r_process_info[FRACTIONAL_STEP];
+    r_process_info.SetValue(FRACTIONAL_STEP, 1);
 
-            for (ModelPart::NodeIterator itNode = NodesBegin; itNode != NodesEnd; ++itNode)
-            {
-                itNode->FastGetSolutionStepValue(REACTION) = ZeroVector(3);
-            }
+#pragma omp parallel for
+        for (int i_node = 0; i_node < static_cast<int>(r_model_part.NumberOfNodes()); ++i_node) {
```

Comment:
Also here we can define number_of_nodes outside the for statement once and use it.